File: C#/ArraysAndHashing/EncodeAndDecodeStrings.cs
public class Solution
{
    public string Encode(IList<string> strs)
    {
        var result = "";

        foreach (var word in strs)
        {
            result += word.Length + "#" + word;
        }

        return result;
    }

    public List<string> Decode(string encodedString)
    {
        var result = new List<string>();
        var index = 0;

        while (index < encodedString.Length)
        {
            var nextHashtag = encodedString.IndexOf('#', index);
            var wordLengthText = encodedString.Substring(
                index,
                nextHashtag - index
            );

            var wordLength = Int32.Parse(wordLengthText);
            var wordStart = nextHashtag + 1;
            var word = encodedString.Substring(wordStart, wordLength);

            result.Add(word);
            index = wordStart + wordLength;
        }

        return result;
    }
}

File: Python/ArraysAndHashing/EncodeAndDecodeStrings.py
class Solution:
    def encode(self, strs: list[str]) -> str:
        result = ""

        for word in strs:
            result += str(len(word)) + "#" + word

        return result

    def decode(self, encoded_string: str) -> list[str]:
        result = []
        index = 0

        while index < len(encoded_string):
            next_hashtag = encoded_string.find("#", index)
            word_length_text = encoded_string[index:next_hashtag]

            word_length = int(word_length_text)
            word_start = next_hashtag + 1
            word = encoded_string[word_start:word_start + word_length]

            result.append(word)
            index = word_start + word_length

        return result
